programs/ssl/ssl_client2.c

```diff
@@ -397,6 +403,77 @@ static int my_verify( void *data, mbedtls_x509_crt *crt, int depth, uint32_t *fl
 }
 #endif /* MBEDTLS_X509_CRT_PARSE_C */
 
+/*
```

Comment:
That's quite a bit of code duplicated between `ssl_server2.c` and `ssl_client2.c`. Should we introduce an `ssl_util2.c`?

Reply:
We already have other duplications between `ssl_server2.c` and `ssl_client2.c`, such as `my_debug`, `my_send` and `my_recv`. If we want to outsource these similarities, I'd prefer we open a separate ticket for it.

Reply:
Ok: IOTSSL-1807